### ISSUE TITLE
[bug] MetaMask wallet does not detect browser plugin but the EIP-6963 detected one works

### ISSUE THREAD

**Issue by KorbinianK** [WalletConnect/web3modal#1981]
### Link to minimal reproducible example

https://stackblitz.com/~/github.com/KorbinianK/svelte-dapp-starter?view=editor

### Summary

I am using the JS version (v4 with wagmi 2).

In the modal, Metamask shows up twice, once because of the EIP-6963 detection and once because in the "featured wallets" section.

<img width="380" alt="image" src="https://github.com/WalletConnect/web3modal/assets/5267230/a0c6fa78-9572-4502-a0e5-2d0d4abbccc1">

The problem now is, when I click the second one (the proper wallet from the explorer), I cannot connect via the browser:

<img width="380" alt="image" src="https://github.com/WalletConnect/web3modal/assets/5267230/b388aa59-bb9f-40c0-9b6d-7bac6a928718">

This can be an issue when users have an outdated metamask version, where the EIP one does not show up and the injected provider might not be configured.

### List of related npm package versions

```json
"@wagmi/connectors": "^4.1.14",
"@wagmi/core": "^2.6.5",
"@web3modal/wagmi": "^4.0.10",
"viem": "~2.7.14"
```

**Comment by KorbinianK** [comment]
I suspect it still expects the `MetaMaskConnector`, which was removed in `wagmi 2.x`

https://wagmi.sh/core/guides/migrate-from-v1-to-v2#removed-metamaskconnector

**Comment by zlccns** [comment]
I also encountered the same situation

**Comment by olegavp** [comment]
Actual too

```
"@wagmi/connectors": "^4.1.14",
"@wagmi/core": "^2.6.5",
"@web3modal/wagmi": "^4.0.14",
"viem": "~2.7.16"
```

**Comment by sh1kxrv** [comment]
i can assume that for some reason web3modal sometimes incorrectly detects the platform and tries to redirect by app deeplink

i patched the [next line](https://github.com/WalletConnect/web3modal/blob/1f5f41b53440fcac6f17b487918e12a0841ea2ca/packages/scaffold/src/views/w3m-connecting-wc-view/index.ts#L131) and the extension was detected again in desktop browser's

```js
// const browserIds = rdns ? [rdns] : injectedIds ?? [] - Original
const browserIds = rdns && CoreHelperUtil.isMobile()? [rdns] : injectedIds ?? [] // Patched
```

maybe this method has pitfalls and can be fixed in a better way

**Comment by sh1kxrv** [comment]
i can assume that for some reason web3modal sometimes incorrectly detects the platform and tries to redirect by app deeplink

i patched the [next line](https://github.com/WalletConnect/web3modal/blob/1f5f41b53440fcac6f17b487918e12a0841ea2ca/packages/scaffold/src/views/w3m-connecting-wc-view/index.ts#L131) and the extension was detected again in desktop browser's

```js
// const browserIds = rdns ? [rdns] : injectedIds ?? [] - Original
const browserIds = rdns && CoreHelperUtil.isMobile() ? [rdns] : injectedIds ?? [] // Patched
```

maybe this method has pitfalls and can be fixed in a better way

**Comment by ibeallc** [comment]
Any solution?

**Comment by DanielSinclair** [comment]
Can confirm that we're also seeing this across the "Browser" tab for every connector, including MetaMask and Rainbow. <img width="390" alt="Screenshot 2024-03-26 at 6 19 45 PM" src="https://github.com/WalletConnect/web3modal/assets/4412473/e3f66d20-cfa0-4af5-b758-750576084010">

**Comment by DanielSinclair** [comment]
Can confirm that we're also seeing this across the "Browser" tab for each connector, including MetaMask and Rainbow. The EIP-6963 connector (when included by the dApp) will function, as will the "Browser Wallet" option.

Reproduction steps:
- Navigate to All Wallets
- Select Rainbow or MetaMask
- Click the Browser tab
- Wallet will not be detected

<img width="391" alt="Screenshot 2024-03-26 at 6 25 40 PM" src="https://github.com/WalletConnect/web3modal/assets/4412473/87ab88f9-3bc3-433f-89e7-d65bac870495">

**Comment by DanielSinclair** [comment]
This looks to be fixed in `4.1.3`
